cmd/dfget/app/root.go

```diff
@@ -275,19 +283,90 @@ func transFilter(filter string) []string {
 	return strings.Split(filter, "&")
 }
 
-func handleNodes() error {
+func handleNodes(nodeWithWeightList []string) ([]string, error) {
 	nodes := make([]string, 0)
+	weightTree := make(map[int][]string)
+	weightKey := make([]int, 0)
 
-	for _, v := range cfg.Nodes {
-		// TODO: check the validity of v.
-		if strings.IndexByte(v, ':') > 0 {
-			nodes = append(nodes, v)
-			continue
+	for _, v := range nodeWithWeightList {
+		node, weight, err := splitNodeAndWight(v)
+		if err != nil {
+			return nil, errors.Wrapf(errortypes.ErrInvalidValue, "node: %s %v", v, err)
 		}
-		nodes = append(nodes, fmt.Sprintf("%s:%d", v, config.DefaultSupernodePort))
+		weightKey = append(weightKey, weight)
+
+		node, err = handleDefaultPort(node)
+		if err != nil {
+			return nil, errors.Wrapf(errortypes.ErrInvalidValue, "node: %s %v", v, err)
+		}
+		weightTree[weight] = append(weightTree[weight], node)
 	}
-	cfg.Nodes = nodes
-	return nil
+
+	// get the the greatest common divisor of the weight slice and
+	gcdNumber := gcdSlice(weightKey)
+	for k, v := range weightTree {
+		for i := 0; i < (k / gcdNumber); i++ {
+			nodes = append(nodes, v...)
+		}
+	}
+
+	return nodes, nil
+}
+
+// gcdSlice returns the greatest common divisor of a slice.
+func gcdSlice(s []int) int {
+	length := len(s)
+	if length == 1 {
+		return s[0]
+	}
+
+	return gcd(s[length-1], gcdSlice(s[:length-1]))
+}
+
+// gcd returns the greatest common divisor of x and y.
+func gcd(x, y int) int {
+	var z int
+	for y != 0 {
+		z = x % y
+		x = y
+		y = z
+	}
+	return x
+}
+
+func handleDefaultPort(node string) (string, error) {
+	result := strings.Split(node, ":")
+	splitLength := len(result)
+
+	if splitLength == 2 {
+		if result[0] == "" || result[1] == "" {
+			return "", errortypes.ErrInvalidValue
+		}
+		return node, nil
+	}
+
+	if splitLength == 1 {
+		return fmt.Sprintf("%s:%d", node, config.DefaultSupernodePort), nil
+	}
+
+	return "", errortypes.ErrInvalidValue
+}
+
+func splitNodeAndWight(value string) (string, int, error) {
+	result := strings.Split(value, "/")
+	splitLength := len(result)
+
+	if splitLength < 1 || splitLength > 2 {
```

Comment:
I think a `switch` here would be more clear, since we're handling only 2 valid options.

```go
switch len(result) {
case 1:
    //
case 2:
    //
default:
    //
}
```